src/numeric/decision_rules.jl

```diff
@@ -1,9 +1,43 @@
-import splines
+abstract AbstractDecisionRule{S,T}
 
-abstract AbstractDecisionRule
-abstract AbtsractCubicSplineDR <: AbstractDecisionRule
+# abstract AbstractCachedDecisionRule{S,T} <: AbstractDecisionRule{S,T}
+# we don't implement that using subtypes anymore
+
+type DecisionRule{S,T} <: AbstractDecisionRule{S,T}
+    grid_exo::S
+    grid_endo::T
+    n_x::Int # number of values (could be a method if there was a nice method name)
+    coefficients::Array{Array{Float64}}
+end
+
+
+type ConstantDecisionRule
+    constants::Vector{Float64}
+end
+
+(dr::ConstantDecisionRule)(x::Vector{Float64}) = dr.constants
+(dr::ConstantDecisionRule)(x::Matrix{Float64}) = repmat(dr.constants',size(x,1),1)
```

Comment:
That's an easy one. To paraphrase Gertrude Stein, "a constant is a constant is a constant" . So no matter where and how you evaluate it, it remains the same.
In general `dr(x)` makes sense for an iid driving process as there is no exogenous process in this case. Or when there is a continuous exognous process you can define without ambiguity `dr([x1;x2])` as `dr(x1,x2)`.